Cargo.toml

```diff
@@ -1,80 +1,112 @@
-primitive-types = { version = "<1", optional = true }
-proptest = { version = "1.0", optional = true }
-pyo3 = { version = "0.18.3", optional = true }
-quickcheck = { version = "1.0", optional = true }
-rand = { version = "0.8.5", optional = true }
-rlp = { version = "0.5.1", optional = true }
-ruint-macro = { version = "1.0.2", path = "./ruint-macro" }
-serde = { version = "1.0", optional = true }
-smallvec = { version = "1.8.0", optional = true, features = ["union"] } # for UintDyn
-# HACK: (BLOCKED) sqlx requires a runtime to be specified.
-# <https://github.com/launchbadge/sqlx/issues/1627>
-sqlx-core = { version = "0.6", optional = true, features = ["runtime-tokio-native-tls"] }
+ruint-macro = { version = "1.0.2", path = "ruint-macro" }
+
 thiserror = "1.0"
-valuable = { version = "0.1.0", optional = true }
-zeroize = { version = "1.5", optional = true }
 
-[build-dependencies]
-# Rust currently has no way to conditionally compile based on compiler features.
-# `autocfg` is a workaround, but doesn't support probing for `#[features(...)]`
-# yet (there's a PR though, but crates.io doesn't allow us to use it).
-# So instead, we do the dumb thing and just check for nightly.
-rustc_version = "0.4.0"
+# support
+alloy-rlp = { version = "0.3", optional = true }
+arbitrary = { version = "1", optional = true }
+ark-ff = { version = "0.4", optional = true }
+bn-rs = { version = "0.2", optional = true }
+fastrlp = { version = "0.3", optional = true }
+num-bigint = { version = "0.4", optional = true }
+parity-scale-codec = { version = "3", optional = true, features = ["derive", "max-encoded-len"] }
+primitive-types = { version = "0.12", optional = true }
+proptest = { version = "1", optional = true }
+pyo3 = { version = "0.19", optional = true }
+quickcheck = { version = "1", optional = true }
+rand = { version = "0.8", optional = true }
+rlp = { version = "0.5", optional = true }
+serde = { version = "1", optional = true }
+valuable = { version = "0.1", optional = true }
+zeroize = { version = "1.6", optional = true }
+
+# postgres
+bytes = { version = "1.4", optional = true }
+postgres-types = { version = "0.2", optional = true }
+
+# sqlx
+sqlx-core = { version = "0.7", optional = true }
 
 [dev-dependencies]
-ruint = { path = ".", features = ["arbitrary", "proptest"] }
+ruint = { workspace = true, features = ["arbitrary", "proptest"] }
 
 criterion = "0.5"
 rand = "0.8"
 
-approx = "0.5.1"
-ark-bn254 = "0.4.0"
+approx = "0.5"
+ark-bn254 = "0.4"
 bincode = "1.3"
 coverage-helper = "0.1"
-hex = "0.4.3"
-hex-literal = "0.4.1"
-postgres = "0.19.3"
-proptest = { version = "1.0" }
+hex = "0.4"
+hex-literal = "0.4"
+postgres = "0.19"
+proptest = "1.2"
 serde_json = "1.0"
 
-[package.metadata.docs.rs]
-rustdoc-args = ["--cfg", "has_cfg_doc", "--html-in-header", ".cargo/katex-header.html"]
-all-features = true
+[features]
+# nightly-only features
+nightly = []
+generic_const_exprs = ["nightly"]
+
+# support
+alloy-rlp = ["dep:alloy-rlp"]
```

Comment:
why are the support features features now explicit?

Reply:
Better to be explicit, and helps in future with chaining on other features that it should enable (eg std, alloc, serde...)